enhancements/installer/aws-custom-dns.md

```diff
@@ -0,0 +1,544 @@
+a completely optional step with this design. If the customer does configure
+their custom DNS solution and leave the in-cluster CoreDNS pod running, all
+in-cluster entities will end up using the CoreDNS pod's DNS services and all
+out of cluster requests will be handled by the external DNS.
+
+### Workflow Description
+
+1. For an OpenShift administrator to use a custom DNS solution, they would
+have to enable a config option in the install configuration. This would
+cause the Installer and the Ingress Operator to skip configuring the cloud
+DNS with entries for API, Internal API and Ingress Services. 
+
+2. The Installer and the Ingress Operator continue to configure the cloud LBs
+on the public subnets of Public Hosted Zones for `api` and `*.apps`. They will
+continue to configure cloud LBs on the private subnets of Private Hosted Zones
+for `api-int` and `*.apps` for in-cluster access to the Ingress service.
+
+3. The Installer will also update the new fields `APIServerDNSConfig` and
+`InternalAPIServerDNSConfig` within the `AWSPlatformStatus`, with the LB
+IP addresses and the DNS Record types for API, API-Int servers respectively.
+
+4. The Installer instructs the machine-config-operator(MCO) to start a static
+CoreDNS pod on the bootstrap and master nodes to provide DNS resolution for
+the API and API-Int servers. MCO uses the DNS config in the `AWSPlatformStatus`
+to generate the CoreFile to configure the CoreDNS instance in the pod.
+Note: This implementation differs from the on-prem case by not staring the
+keepalived or haproxy containers for load balancing.
+
+5. The Installer omits the DNS Zones within `dnses.config.openshift.io/cluster`
+to influence the behavior of the Ingress Operator by making it skip the
+configuration of cloud DNS while continuing with LB configuration.
+
+6. The Ingress Controllers managed by the cluster-ingress-operator continue
+to configure the application LBs to enable in-cluster and external accesss to
+cluster services. The Ingress Controllers will continue to update the DNSRecord
+ConfigResource(CR) with the configuration required to configure DNS for these
+services.
+
+7. The MCO will watch the DNSRecord CR for the DNS configuration mentioned in
+step 6. MCO will update the newly added `IngressDNSConfig` field within the
+`AWSPlatformStatus` with the DNS configuartion for *.apps records by reading
+the DNSRecord CR.
+
+8. The runtimecfg utility will monitor the `Infrastructure` CR for changes to
+`IngressDNSConfig` of `AWSPlatformStatus` to update the CoreDNS pod's
+CoreFile to now also include DNS configuration for both internal and
+external access to the cluster services (the `*.apps` records).
+
+9. Once the cluster is functional, the user can look at the contents of the new
+fields `APIServerDNSConfig` and `IngressDNSConfig` in the `AWSPlatformStatus`
+field of the `Infrastructure` CR to configure their external DNS with entries
```

Comment:
The installer would be declaring install complete, for a cluster that still needs post-install configuration. Generally we try to validate as much as possible in advance to ensure the install will succeed. We can't do that if DNS is configured post-install. Are there any support considerations there?

Reply:
The CoreDNS pod running within the cluster will make sure API and Ingress services are available for Install to declare "Install Complete".
We are asking the customer to configure their external DNS solution post-install to handle API and apps requests coming from outside the cluster.